crates/ruff_server/src/server.rs

```diff
@@ -252,6 +255,20 @@ impl Server {
                 },
             )),
             hover_provider: Some(types::HoverProviderCapability::Simple(true)),
+            notebook_document_sync: Some(types::OneOf::Left(NotebookDocumentSyncOptions {
+                save: Some(false),
+                notebook_selector: [NotebookSelector::ByNotebook {
+                    notebook: types::Notebook::NotebookDocumentFilter(
+                        types::NotebookDocumentFilter::ByType {
+                            notebook_type: "jupyter-notebook".to_string(),
+                            scheme: Some("file".to_string()),
+                            pattern: None,
+                        },
```

Comment:
Is this specific to VS Code? I'm referring to the "jupyter-notebook" type. For reference, we don't use any notebook document filter in `ruff-lsp`:

https://github.com/astral-sh/ruff-lsp/blob/b8a48009f35756dc51268beb6bbc39c21a1e3a46/ruff_lsp/server.py#L129-L140

This asks to only send the request to sync the Python code cells as we aren't interested in any kind of cell.

Reply:
I got this from the LSP specification:
<img width="664" alt="Screenshot 2024-05-15 at 9 37 34 PM" src="https://github.com/astral-sh/ruff/assets/19577865/62f64858-46ed-4664-a8a3-b4be1a936012">

I assume this would be the same across editors.

Reply:
I see. I'm not sure where does the `jupyter-notebook` type comes from.

> I assume this would be the same across editors.

I think it depends - If it is set by VS Code, then I don't think it would be same.

I don't think we need to worry about this now but, just to be on the safer side, I'd probably keep this same as `ruff-lsp`.

Does this work on unsaved Notebook files? Like, a new notebook file is opened in VS Code but it doesn't exists on disk yet.

Reply:
> Does this work on unsaved Notebook files? Like, a new notebook file is opened in VS Code but it doesn't exists on disk yet.

This does fail, but only because new notebook cell url gets passed into `didOpenTextDocument`, which isn't expected. Do we need to support 'unattached' notebook cells?

Reply:
Hmmm.... it seems like, in general, that the server doesn't support files created from `Ctrl+N` in VS Code if they aren't saved to the file system. I'll create a follow-up issue for this.

Reply:
> This does fail, but only because new notebook cell url gets passed into `didOpenTextDocument`, which isn't expected. Do we need to support 'unattached' notebook cells?

I just checked, we do support it. It's probably because we never touch the filesystem in the new server and always work with stdin with `ruff` command.